### PR TITLE
Fix a syntax error

### LINKED ISSUE #7: Bug: error thrown when dispatching an `Update()`
**Type:** Bug
**Library version:** 1.0.0

**What's happening?**

I'm creating a reducer and dispatching an `Update()`. This is throwing the following error:

> index.js:70 Uncaught TypeError: newSideEffect is not iterable

This is the [LoC throwing the error](https://github.com/conorhastings/use-reducer-with-side-effects/blob/8757c2bae69d02663d3d36806aae668e3c39b405/src/index.js#L37-L41)

**What's the expected behavior?**

This would not error.

**What's the source of the problem?**

I believe this problem was introduced in eac6b68efa53f9 and then almost but not quite fixed in 54e75bf342 . Currently, [the LoC that are throwing the error](https://github.com/conorhastings/use-reducer-with-side-effects/blob/8757c2bae69d02663d3d36806aae668e3c39b405/src/index.js#L37-L41) don't do anything.

**Possible resolution**

There are a bunch of ways to fix this, one is:

```js
const newSideEffects = newSideEffect
      ? [
          ...state.sideEffects,
          ...(Array.isArray(newSideEffect) ? newSideEffect : [newSideEffect]),
        ]
      : state.sideEffects;
```

### PR DESCRIPTION
**Pull request by jamesplease**
Resolves gh-7

---

This is one way to resolve #7 , but feel free to close this out if you'd prefer to write it another way.

### CODE REVIEW

**Comment by conorhastings** [comment]
thanks, though ti caught hat, need to add tests

File: src/index.js [hunk 1]
```diff
@@ -33,11 +33,11 @@ function finalReducer(reducer) {
...
     let { newState, newSideEffect } = reducer(state.state, action);
-    const newSideEffects = Array.isArray(newSideEffect) ? newSideEffect : [newSideEffect];
-       newSideEffects && newSideEffects.length ? [
-        ...state.sideEffects,
-        ...newSideEffect
-      ]
+    const newSideEffects = newSideEffect
+      ? [
+          ...state.sideEffects,
+          ...(Array.isArray(newSideEffect) ? newSideEffect : [newSideEffect]),
```
Comment (conorhastings) [inline src/index.js:39]: ahh yup this is what i missed

**Comment by conorhastings** [comment]
published in `v1.0.1`

**Comment by jamesplease** [comment]
Thank you! 🙏

**Pull request merged**